dot-net-sdk/store/ConfigurationStore.cs

```diff
@@ -26,51 +27,89 @@ public class ConfigurationStore : IConfigurationStore
         _banditFlagCache = banditFlagCache;
     }
 
-    public void SetFlag(string key, Flag flag)
+    private void SetFlag(string key, Flag flag)
     {
         _flagConfigurationCache.Set(key, flag, new MemoryCacheEntryOptions().SetSize(1));
     }
 
-    public void SetBanditModel(Bandit banditModel)
+    private void SetBanditModel(Bandit banditModel)
     {
         _banditModelCache.Set(
             banditModel.BanditKey,
             banditModel,
             new MemoryCacheEntryOptions().SetSize(1));
     }
 
-    public void SetBanditFlags(BanditFlags banditFlags)
+    private void SetBanditFlags(BanditFlags banditFlags)
     {
         _banditFlagCache.Set(BANDIT_FLAGS_KEY, banditFlags, new MemoryCacheEntryOptions().SetSize(1));
     }
 
     public BanditFlags GetBanditFlags()
     {
-        if (_banditFlagCache.TryGetValue(BANDIT_FLAGS_KEY, out BanditFlags? banditFlags) && banditFlags != null)
+        lock (Baton)
```

Comment:
This will not allow reads if a write is happening?

Reply:
Did a little more research and found `ReaderWriterLock` which allows for concurrent reading and exclusive writing, so we get to block reads while the cache is updating, but allow all the threads to read any of the stores concurrently!